birdhouse/components/stac/config/magpie/stac_hooks.py

```diff
@@ -0,0 +1,144 @@
+    """
+    Create the stac item resource
+    """
+    request = response.request
+    body = request.json
+    item_id = body['id']
+    display_name = extract_display_name(body['links'])
+
+    # Get the <collection_id> from url -> /collections/{collection_id}/items
+    collection_id = re.search(r'(?<=collections\/).*?(?=\/items)', request.url).group()
+
+    # note: matchdict reference of Twitcher owsproxy view is used, just so happens to be same name as Magpie
+    service = get_service_matchdict_checked(request)
+    try:
+        # Getting a new session from the request
+        session = get_session_from_other(request.db)
+        children = ru.get_resource_children(service, db_session=session, limit_depth=5)
+        item_res_create = True
+        # find "stac/stac"
+        for childs in children.values():
+            if childs["node"].resource_name == "stac":
+                # find "stac/stac/collections"
+                for stac_child in childs["children"].values():
+                    if stac_child["node"].resource_name == "collections":
+                        # find the id of the resource matching stac/stac/collections/<collection_id>
+                        for collection_child in stac_child["children"].values():
+                            if collection_child["node"].resource_name == collection_id:
+                                collection_res_id = collection_child["node"].resource_id
+                                # find the id of the resource matching stac/stac/collections/<collection_id>/items
+                                for item_child in collection_child["children"].values():
+                                    if item_child["node"].resource_name == "items":
+                                        item_res_id = item_child["node"].resource_id
+                                        item_res_create = False
+                                        break
+
+        # create resource /stac/stac/collections/<collection_id>/items if does not already exist
+        if item_res_create:
+            item_res = ru.create_resource("items", None, Route.resource_type_name, collection_res_id, db_session=session)
+            item_res_id = item_res.json["resource"]["resource_id"]
+
+        # create resource stac/stac/collection/<collection_id>/items/<item_id>
+        # In try catch since Magpie resource are persistent but stac-db could be wiped
+        try:
+            item_res = ru.create_resource(item_id, display_name, Route.resource_type_name, item_res_id, db_session=session)
+        except Exception as exc:
+            LOGGER.warning("Failed creation of the item %s %s", display_name, str(exc), exc_info=exc)
+        session.commit()   
+   
+    except Exception as exc:
+        LOGGER.error("Unexpected error while creating the item %s %s", display_name, str(exc), exc_info=exc)
+    
+    return response
+
+def extract_display_name(links):
+    # type: (JSON array) -> string
```

Comment:
Shouldn't this be: `# type: List[Dict] -> Optional[str]`

Reply:
yes you are right thanks

Reply:
`# type: (List[Dict[str, str]]) -> Optional[str]` 

But I would prefer `-> str` with raise error as per https://github.com/bird-house/birdhouse-deploy/pull/386#discussion_r1341407674